agent/provider/src/provider_agent.rs

```diff
@@ -66,6 +67,56 @@ impl GlobalsManager {
     }
 }
 
+struct WhitelistManager {
+    state: DomainWhitelistState,
+    monitor: Option<FileMonitor>,
+}
+
+impl WhitelistManager {
+    fn try_new(whitelist_file: &Path) -> anyhow::Result<Self> {
+        let mut patterns = DomainPatterns::load_or_create(whitelist_file)?;
+        patterns.update_and_save(whitelist_file, patterns.clone())?; //TODO fix ownership issue
```

Comment:
Why are you updating the `patterns` variable with itself and saving it instead of just doing `patterns.save()?`?

Reply:
I guess I needed it at some point, but not anymore. I runned tests and have checked feature manually after removing it and works.
Done.